File: <100/071/071.jl
function solve(upper_limit :: Int) :: Int
  s :: Set{Rational{Int}} = Set()
  for d ∈ 9:upper_limit
    n :: Int = 3 * d ÷ 7
    while 1 < n && gcd(n, d) != 1
      n -= 1
    end
    if 0 < n
      push!(s, n//d)
    end
  end

  return s |> maximum |> numerator
end


function main() :: Nothing
  upper_limit = 10^6
  upper_limit |> solve |> println
end


@time main()

File: <100/071/071.py
from fractions import Fraction
import math
import time


def main() -> int:
    upper_limit = 10**6
    l: list[Fraction] = []
    for d in range(9, upper_limit+1):
        n = 3 * d // 7
        while 1 < n and math.gcd(n, d) != 1:
            n -= 1
        if 0 < n:
            l.append(Fraction(n, d))
    
    l = sorted(l)
    return l[-1].numerator


if __name__ == "__main__":
    start = time.time()
    answer = main()
    elapsed = time.time() - start

    print(f'{elapsed} sec')
    print(answer)
